library/pkparse.c

```diff
@@ -667,7 +830,22 @@ int mbedtls_pk_parse_subpubkey(unsigned char **p, const unsigned char *end,
 #endif /* MBEDTLS_RSA_C */
 #if defined(MBEDTLS_ECP_LIGHT)
     if (pk_alg == MBEDTLS_PK_ECKEY_DH || pk_alg == MBEDTLS_PK_ECKEY) {
-        ret = pk_use_ecparams(&alg_params, &mbedtls_pk_ec(*pk)->grp);
+        switch (ec_grp_id) {
+#if defined(MBEDTLS_ECP_DP_CURVE25519_ENABLED)
```

Comment:
I appreciate your care for minimizing code size, but I find the resulting code to a be bit too cluttered. Also, I think it would be good to have the list of curves only in a single place - both macros and identifiers. Suggestion:
```c
// some place near the top of the file (or even shared with pkwrite.c)
#if defined(MBEDTLS_ECP_DP_CURVE25519_ENABLED) || defined(MBEDTLS_ECP_DP_CURVE448_ENABLED)
#define MBEDTLS_PK_HAVE_RFC8410_CURVES
#endif

#if defined(MBEDTLS_PK_HAVE_RFC8410_CURVES)
static int pk_is_rfc8410_curve(mbedtls_ecp_group_id id)
{
#if defined(MBEDTLS_ECP_DP_CURVE25519_ENABLED)
    if (id == MBEDTLS_ECP_DP_CURVE25519)
        return 1;
#endif
#if defined(MBEDTLS_ECP_DP_CURVE448_ENABLED)
    if (id == MBEDTLS_ECP_DP_CURVE448)
        return 1;
#endif
    return 0;
}
#endif

// usage
#if defined(MBEDTLS_PK_HAVE_RFC8410_CURVES)
if (pk_is_rfc8410_curve(ec_grp_id)) {
    // this is a curdle curve
}
else
#endif
{
    // non-curdle curve
}
```
Wdyt?